### PR TITLE
  TI-2402 - User list Filter

### PR DESCRIPTION
**Pull request by stewartm83**
Please review.

### CODE REVIEW

File: db.go [hunk 1]
```diff
@@ -24,6 +24,13 @@ var (
 	nextUserId    UserId
 )
 
+type GetIdentitiesFlag int
+
+const (
+	GetIdentitiesFlagNone    GetIdentitiesFlag = 1 << iota
```
Comment (bmharper) [inline db.go]: The none flag must be zero. Simplest thing I can come up with is this:
```go
const (
    GetIdentitiesFlagNone GetIdentitiesFlag = 0
    GetIdentitiesFlagDeleted   = 1 << (iota - 1)
)
```

File: db.go [hunk 2]
```diff
@@ -287,15 +295,16 @@ func (x *dummyUserStore) RenameIdentity(oldEmail, newEmail string) error {
...
-func (x *dummyUserStore) GetIdentities() ([]AuthUser, error) {
+func (x *dummyUserStore) GetIdentities(getIdentitiesFlag GetIdentitiesFlag) ([]AuthUser, error) {
 	x.usersLock.RLock()
 	defer x.usersLock.RUnlock()
 
 	list := []AuthUser{}
 	for _, v := range x.users {
-		if v.archived == false {
-			list = append(list, AuthUser{v.userId, v.email, v.username, v.firstname, v.lastname, v.mobilenumber, v.telephonenumber, v.remarks, v.created, v.createdby, v.modified, v.modifiedby, v.authUserType})
+		if getIdentitiesFlag != GetIdentitiesFlagDeleted && v.archived == true {
```
Comment (bmharper) [inline db.go]: This is a flag, so always use a binary AND to check for the presence of a flag.

File: db_sql.go [hunk 3]
```diff
@@ -339,8 +339,14 @@ func (x *sqlUserStoreDB) RenameIdentity(oldIdent, newIdent string) error {
 	}
 }
 
-func (x *sqlUserStoreDB) GetIdentities() ([]AuthUser, error) {
-	rows, err := x.db.Query("SELECT userid, email, username, firstname, lastname, mobile, phone, remarks, created, createdby, modified, modifiedby, authusertype FROM authuserstore WHERE (archived = false OR archived IS NULL)")
+func (x *sqlUserStoreDB) GetIdentities(getIdentitiesFlag GetIdentitiesFlag) ([]AuthUser, error) {
+	var sqlStatement string
+	if getIdentitiesFlag == GetIdentitiesFlagDeleted {
```
Comment (bmharper) [inline db_sql.go]: Use binary and to check for field presence.

Also, make the common part of the query a common string. And only add that final "WHERE" clause if the flag is not present.

Comment (RoanBrand) [reply db_sql.go]: `if (getIdentitiesFlag && GetIdentitiesFlagDeleted) `

Comment (bmharper) [reply db_sql.go]: Don't listen to him!
`if getIdentitiesFlag & GetIdentitiesFlagDeleted !=0`

**Comment by bmharper** [comment]
Looks good

**Pull request merged**